pkg/util/limiter/query_limiter_test.go

```diff
@@ -0,0 +1,91 @@
+package limiter
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/cortexproject/cortex/pkg/cortexpb"
+)
+
+func TestPerQueryLimiter_AddFingerPrint(t *testing.T) {
+	const (
+		metricName = "test_metric"
+	)
+
+	var (
+		series1 = labels.FromMap(map[string]string{
+			labels.MetricName: metricName + "_1",
+			"series1":         "1",
+		})
+		series2 = labels.FromMap(map[string]string{
+			labels.MetricName: metricName + "_2",
+			"series2":         "1",
+		})
+		matchers = []*labels.Matcher{
+			labels.MustNewMatcher(labels.MatchEqual, labels.MetricName, metricName),
+		}
+
+		limiter = NewQueryLimiter(100)
+	)
+	limiter.AddSeries(cortexpb.FromLabelsToLabelAdapters(series1), matchers)
+	err := limiter.AddSeries(cortexpb.FromLabelsToLabelAdapters(series2), matchers)
+	assert.Equal(t, 2, limiter.UniqueSeries())
+	assert.Nil(t, err)
```

Comment:
A couple of thing about error checking:
1. Always check "no error as first thing"
2. Use `require.NoError()` so that test execution stops in case of unexpected error

```suggestion
	require.NoError(t, err)
	assert.Equal(t, 2, limiter.UniqueSeries())
```